src/services/createPropStore.ts

```diff
@@ -2,43 +2,61 @@ import { InjectionKey, reactive } from 'vue'
 import { isWithComponent, isWithComponents } from '@/types/createRouteOptions'
 import { ResolvedRoute } from '@/types/resolved'
 import { Route } from '@/types/route'
-import { MaybePromise } from '@/types/utilities'
 
 export const propStoreKey: InjectionKey<PropStore> = Symbol()
 
 type ComponentProps = { id: string, name: string, props?: (params: Record<string, unknown>) => unknown }
 
 export type PropStore = {
-  setProps: (route: ResolvedRoute) => void,
-  getProps: (id: string, name: string, params: unknown) => MaybePromise<unknown> | undefined,
+  setProps: (route: ResolvedRoute, options: { prefetch: boolean }) => void,
+  getProps: (id: string, name: string, params: Record<string, unknown>) => unknown,
 }
 
 export function createPropStore(): PropStore {
-  const store = reactive(new Map<string, unknown>())
+  const store: Map<string, unknown> = reactive(new Map())
 
-  function setProps(route: ResolvedRoute): void {
-    store.clear()
+  const setProps: PropStore['setProps'] = (route, { prefetch }) => {
+    const routeKeys: string[] = []
+    const componentProps = route.matches.flatMap(match => getComponentProps(match))
 
-    route.matches
-      .flatMap(match => getComponentProps(match))
-      .forEach(({ id, name, props }) => {
-        if (props) {
-          const key = getPropKey(id, name, route.params)
-          const value = props(route.params)
+    for (const { id, name, props } of componentProps) {
+      if (!props) {
+        continue
+      }
 
-          store.set(key, value)
-        }
-      })
+      const key = getPropKey(id, name, route.params)
+      routeKeys.push(key)
+
+      if (store.has(key)) {
+        continue
+      }
```

Comment:
I'm not 100% sure skipping every time is the right thing to do here. But I think it makes sense?